### ISSUE TITLE
I need 1280*1280 image size pre-training model 

### ISSUE THREAD

**Issue by zxt243416724** [ultralytics/ultralytics#719]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

I need a pre-training model with an image size of 1280, because my project has a large image with a resolution size of 4096 * 4096. The default size of 640 is not ideal.

I tried to train myself from scratch. Without powerful hardware and few sample pictures, I couldn't converge.

### Use case

no 

### Additional

no

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by AyushExel** [comment]
Yeah, we have a similar hardware problem. We'll release the the 1280 model in the 8.1 release

**Comment by JustasBart** [comment]
@AyushExel Sounds really good! :rocket:

**Comment by purple-mesh** [comment]
Hi @AyushExel, just have a questions regarding the various models available now.

I see that there is `ultralytics/models/v8/yolov8x6.yaml` model which is both defined and available for download. Is this what you were referring to as the "1280 model" or you were referring to another completely different one that is work-in-progress?

**Comment by AyushExel** [comment]
Yes, its the same model. The official weights are not available to download. the one you might get won't have the best performance.

**Comment by purple-mesh** [comment]
Thanks for the clarification @AyushExel !

So if all works out :crossed_fingers: we should get another updated weights with the best performance in version 8.1 release?

Really looking forward to it.

**Comment by bit-scientist** [comment]
Hi, @AyushExel, could you provide an approximate date for better models release? And when are you planning to share the models trained with 1280? Thank you.

**Comment by AyushExel** [comment]
Hey 1280 models are not a priority right now for us. We've launched the new yolov5u models with p6 versions. The v8 p6 models don't have a timeline yet because we shifted focus to supporting keypoints/pose task.

**Comment by eVen-gits** [comment]
@AyushExel thank you for your hard work.

I'm wondering if there has been any progress on developing larger models for ultralytics' YOLO implementation? Currently, I am working with images that are around 11.664 x 8.750 in size, some even larger, and the 640x640 model struggles to detect elements with high aspect ratios.

I've been using the "sahi" project on GitHub, but even then some elements are too large to fit in a window and the model has trouble detecting them. I think this may be related to the anchor system, but I'm not sure.

As it stands, using the full image that gets scaled down is not an option since some elements are too small for the 640x640 model to detect.

I'm hoping that there will be an update on the development of 1280x1280 models, as this would be of great help. In the meantime, I would encourage anyone interested to check out the "sahi" project on GitHub.

Thank you.

**Comment by glenn-jocher** [comment]
@eVen-gits hello! Thank you for your feedback and for using our YOLO implementation. We are aware that some users require larger models, and we are exploring ways to optimize the performance of our existing models for higher aspect ratio images. However, we do not have an immediate timeline for the release of larger models at the moment. In the meantime, we recommend experimenting with different anchor settings to see if that improves detection on your images. We appreciate your continued support and thank you for your patience.

**Comment by sadimoodi** [comment]
following, appreciate having larger models that support 1280 image size

**Comment by glenn-jocher** [comment]
@sadimoodi, thank you for your interest in Ultralytics YOLO. We understand that larger models that support 1280 image size are important for many users and we appreciate your request. At Ultralytics, we are committed to optimizing our existing models to improve their performance on high aspect ratio images. While we do not have a specific timeline for the release of larger models that support 1280 image size at this time, we are aware of this need and are exploring ways to provide solutions that address this requirement. As always, we appreciate your support and will continue to provide updates as developments take place.

**Comment by taesanbeck** [comment]
Hi Glenn,

I am a data analytics engineering student in my last semester and I am working with 3 meter resolution imagery from worldview satellites. We have a custom dataset created in RoboFlow, 2 classes, Damaged or Undamaged buildings from 7 different cities in Ukraine and we are very interested in you potentially releasing your xView object section weights in .pt format and a 1280 image size yolov8. This would be a game changer for many of us. Thanks for what you do man, Ultralytics is awesome!

Respectfully,
SB

**Comment by glenn-jocher** [comment]
@taesanbeck hello SB,

It's great to hear about your project and the impactful work you are doing with satellite imagery analysis. We are honored that our technology could potentially contribute to such important tasks as assessing building damage.

Regarding the xView weights, our current focus is on improving our algorithms and models to support a wider range of tasks and data resolutions. While we don't have released .pt format weights specific for xView at this moment, we recognize the value this would provide to the community.

As for the 1280 image size model for YOLOv8, we understand the demand for higher resolution models that can handle larger images while maintaining high accuracy on small objects. While we do not have an immediate update on the release of a dedicated 1280 model, we encourage you to stay tuned to our updates. We are always working to meet the community's needs and will make announcements when new models and updates are available.

Keep an eye on our GitHub repository and documentation for the latest updates. We sincerely appreciate the recognition and support of Ultralytics!

Warm regards to you and your academic endeavors in data analytics engineering. Keep up the excellent work!

**Comment by JustasBart** [comment]
@taesanbeck Just to be clear, you are aware that you can train any size model starting from the 640x640 trained model, right? Is it just taking too long for your model to generalize? One other trick you could do is to train intermediate models, such as 1280x1280, then maybe ramp it up to 1920x1920, etc... Also, I hope that you're aware that these YOLO models can also be based on a rectangular image for example 1920x1056 (For a 32 stride network).

All the best! 🚀

**Comment by taesanbeck** [comment]
@JustasBart I did see the strides of 32 will work in other posts. Thanks for the reply though and the active conversation on the topic. Before I posted above to @glenn-jocher I had tried 640 x 640 but like mentioned above it would only detect larger buildings and not the smaller ones, I had also tried 1280 x 1280 which did slow down the a bit during training but seemed to improve the mAP50 score quite a bit. Our biggest issue at the moment is our dataset is relatively small, we took several very large images from world view3 satellites from Maxar about 76 images, we have an imagery analyst on our team who has annotated 20,000 annotation over 100 hours of work, he has trained one other person on the team and we are continuing to annotate and retrain. Our initial Idea was to get the smaller dataset to a mAP50 score of around 30 to 40% and the apply transfer learning from a larger model trained on more data like xView, DOTA, etc... But we started to realize DOTA weights were classification not object detection and we're from a resnet50 architecture so those didn't work, xView was ideal, but I must have played with glenns yolov5 repo and xView yaml file for about a month and I could never get the geojson to split the annotations into individual .txt files. The code in the yaml seemed to make two auto split.txt files with all the names of the annotations inside of each file. I am also in 2 other classes this semester so we kind of just said we will just annotated as much as possible and see how it improves. I have also been throttled on google colab even though I am a paying colab user and our free version of roboflow is struggling. So far the best mAP50 we are getting is 36% but it is difficult to get the model to generalize. We are pertaining on yolov8n, the larger models pretrained on COCO actually make the satellite model worse, but nano does the best so far.

**Comment by taesanbeck** [comment]
I also forgot to mention we are splitting those 76 large images during preprocessing. They are tiled 14 x 14 into smaller images and we are gradually increasing image size in strides of 32, last successful run was 1280 x 1280. We have also found that augmentation is good but in small doses because the imagery is already 3m resolution so we tilted +-1 degree, and a few others, but really found better results with minimal preprocessing and increasing image size. Thanks for the conversation from the community, it is very helpful. -SB

**Comment by glenn-jocher** [comment]
@JustasBart, your approach to data preparation and augmentation is sound, especially given the unique challenges of satellite imagery analysis. As you correctly pointed out, excessive augmentation, particularly with rotations, can negatively affect label accuracy in object detection tasks. This is an important consideration to account for when dealing with bounding box annotations.

Incrementally increasing image size is a good strategy for training neural networks on higher resolution imagery, especially when combined with careful augmentation. The fact that you're seeing improved results with minimal preprocessing aligns with the notion that high-quality, well-annotated data often trumps complex data augmentation methods, particularly in the domain of high-resolution image analysis.

It's excellent that you are considering transfer learning from related datasets. However, as you've discovered, the compatibility of weights across different tasks and architectures can indeed be an obstacle. YOLOv5 and YOLOv8 are designed for object detection, and combining their strengths with a carefully annotated dataset like yours should yield increasingly promising results as the dataset grows.

Continued dedication to meticulous annotation, consideration of relevant transfer learning opportunities, and ongoing experimentation with network configuration and image sizes will serve your project well. The community appreciates your detailed sharing of your process and challenges, and we all learn from such exchanges.

Keep up the fantastic work, and feel free to continue the conversation as your project evolves.

Good luck with your ongoing training and research, and may your mAP scores reflect the hard work of your team! 🌟

**Comment by afonsol8** [comment]
Is the pretrained 1280x1280 model out now?

**Comment by glenn-jocher** [comment]
@afonsol8 no, a pretrained 1280x1280 model is not currently available. Please stay tuned to our repository for future updates. 🚀

**Comment by kubatom** [comment]
Are there any updates on the 1280 model? Is this something that you're actively working on?

**Comment by glenn-jocher** [comment]
@kubatom hi there! 🌟

As of now, we don't have a pretrained 1280x1280 model available, and it's not in our immediate roadmap. However, we're continually updating and improving our models, so stay tuned! In the meantime, you can train a model with larger image sizes starting from our 640x640 models using the techniques discussed in our documentation.

For example:
```bash
yolo detect train data=your_dataset.yaml model=yolov8n.pt imgsz=1280
```

Thanks for checking in and for your interest in the project!
